### PR TITLE
Merge fill_status_cache into monitor and split out differently

### PR DESCRIPTION
**Pull request by coolo**
This removes quite some nonsense that the code collected over the years

### CODE REVIEW

File: src/api/app/controllers/webui/project_controller.rb [hunk 1]
```diff
@@ -606,52 +558,86 @@ def unlock
...
+
+    find_opt = { project: @project, view: 'status', code: @status_filter,
+                 arch: @arch_filter, repository: @repo_filter }
+    find_opt[:lastbuild] = 1 if @lastbuild_switch.present?
+
+    buildresult = Buildresult.find_hashed(find_opt)
+    if buildresult.empty?
+      flash[:warning] = "No build results for project '#{@project}'"
+      redirect_to action: :show, project: params[:project]
+      return
```
Comment (Ana06) [inline src/api/app/controllers/webui/project_controller.rb]: This is now only returning to the previous method and afterward `@buildresult_unavailable = true` is set although you are going to redirect... :thinking:

File: src/api/app/controllers/webui/project_controller.rb [hunk 2]
```diff
@@ -606,52 +558,86 @@ def unlock
...
+    end
+
+    return unless buildresult.key?('result')
+    buildresult
+  end
+
+  def monitor_parse_buildresult(buildresult)
+    @packagenames = Set.new
```
Comment (Ana06) [inline src/api/app/controllers/webui/project_controller.rb]: why a set? before we were not taking care of repetitions and sorting is less efficient now as you need to convert to an array....

File: src/api/app/controllers/webui/project_controller.rb [hunk 3]
```diff
@@ -387,68 +387,20 @@ def move_path
...
-    fill_status_cache
+    monitor_parse_buildresult(buildresult)
 
-    @localpackages = {}
-    @project.packages.each do |package|
-      @localpackages[package.name] = 1
+    # extract repos
+    repohash = {}
+    @statushash.each do |repo, arch_hash|
+      repohash[repo] = arch_hash.keys.sort!
```
Comment (Ana06) [inline src/api/app/controllers/webui/project_controller.rb]: This is less efficient than before... you need to iterate once to get the keys and another onw to sort... and you are using this only twice... `sort!` in the first iteration it is more efficient

File: src/api/app/controllers/webui/project_controller.rb (hunk 1 above)
Comment (coolo) [reply src/api/app/controllers/webui/project_controller.rb]: Does it harm?

File: src/api/app/controllers/webui/project_controller.rb (hunk 2 above)
Comment (coolo) [reply src/api/app/controllers/webui/project_controller.rb]: we were taking care of repetitions. This set replaces

```
    @packagenames = @packagenames.flatten.uniq.sort!

    ## Filter for PackageNames ####
    @packagenames.select! { |name| filter_matches?(name, @name_filter) } if @name_filter.present?

File: src/api/app/controllers/webui/project_controller.rb (hunk 3 above)
Comment (coolo) [reply src/api/app/controllers/webui/project_controller.rb]: What first iteration? the arch_hash is a hash and the archs are its keys.

File: src/api/app/controllers/webui/project_controller.rb (hunk 1 above)
Comment (Ana06) [reply src/api/app/controllers/webui/project_controller.rb]: no, but it is not very elegant... let's say it is a nitpick...

**Pull request merged**